modules/samtools/convert/main.nf

```diff
@@ -14,18 +13,27 @@ process SAMTOOLS_BAMTOCRAM {
     path  fai
 
     output:
-    tuple val(meta), path("*.cram"), path("*.crai"), emit: cram_crai
-    path  "versions.yml"                           , emit: versions
+    tuple val(meta), path("*.cram"), path("*.crai") , emit: cram_crai, optional: true
```

Comment:
will fix the meta.yml

Reply:
I agree with the need for explicit output names, but I also like to avoid extra clutter. Since there's only one output and dev's are likely to know which direction you're converting, it makes sense IMO to throw them together.

Reply:
on trying this I am not sure a glob pattern will be much easier to read. 
We would definitely want to keep a fixed order of the elements. so either we could go with something like `*.am` and `*.ai` , which I don't find more readable or probably come up with some regex with `path("${prefix}")` 

I would not like to rename it to `reads_index` because reads are usually associated with Fastq file IMO.

Reply:
What about:
```
tuple val(meta), path("*.{bam,cram}"), emit: alignment
tuple val(meta), path("*.{bai,crai}"), emit: index
 ```

Reply:
ooh, I like that, but that changes the output format, compared to the previous version

Reply:
I kind of like the tuple output includes the index, this avoids me having to merge them later on.
also: +1 on the naming!

Reply:
updated